core/src/main/java/io/opentelemetry/android/instrumentation/AndroidInstrumentation.kt

```diff
@@ -30,11 +28,7 @@ interface AndroidInstrumentation {
      * only be called from [OpenTelemetryRum]'s builder once the [OpenTelemetryRum] instance is initialized and ready
      * to use for generating telemetry.
      *
-     * @param application The Android application being instrumented.
-     * @param openTelemetry The [OpenTelemetry] instance to use for creating signals.
+     * @param ctx The InstallationContext under which the instrumentation is being installed.
      */
-    fun install(
-        application: Application,
-        openTelemetry: OpenTelemetry,
-    )
+    fun install(ctx: InstallationContext)
```

Comment:
I like this idea. I still need to take a deeper look at the whole PR so I can provide a better review overall, though this part reminded me of [this other PR](https://github.com/open-telemetry/opentelemetry-android/pull/661) which makes it a bit confusing to me because there it seemed like we wanted to avoid passing core-specific types to instrumentations? I'm ok with doing so btw, I just want to make sure what's the overall consensus of the kind of use cases we want to support for instrumentations (whether we need them to be fully independent or not).

Reply:
And when you say "core-specific types" in this case you mean both the `InstallationContext` itself but also the `ServiceManager` contained within?

In any case, the api needs to have _something_ to pass to `install()`, and instrumentations might want to leverage the consistent way we've started setting up to get at services. I'm open to other ways that don't involve global/static/singletons. :)

Reply:
> And when you say "core-specific types" in this case you mean both the `InstallationContext` itself but also the `ServiceManager` contained within?

Yes, that's what I was thinking.

> In any case, the api needs to have something to pass to install(), and instrumentations might want to leverage the consistent way we've started setting up to get at services. I'm open to other ways that don't involve global/static/singletons. :)

Makes sense, we still need to pass something to instrumentations, and I like this idea of passing a single object of ours because it means that we could add more tools in the future if needed without having to change the signature of `install()`.

I think the solution to this will become clearer once we fully understand the problem, if it's a matter of having "zero dependency" on anything that's outside of the upstream java repo, then things might get quite complicated, however, if it's just a matter of package size concerns and having a lot of code from a dependency that might not be fully used, such as the Android core, then it's probably a matter of extracting the instrumentation-related files out of core, as @surbhiia mentioned [here](https://github.com/open-telemetry/opentelemetry-android/issues/658), where we can also add a new type like `InstallationContext`. If I understood correctly, I think the problem is the latter, so maybe we can get away with just moving this type into the separate "instrumentation API" module. Now, what to do about the service manager is another issue though, I hope we can discuss the details properly in a SIG meeting.